File: backend/app/schemas/employee.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime, date


class EmployeeCreate(BaseModel):
    """建立員工"""
    employee_no: str = Field(..., min_length=1, max_length=50, description="員工編號")
    employee_type: str = Field(..., description="員工類型 (admin/full_time/part_time/intern)")
    name: str = Field(..., min_length=1, max_length=100, description="姓名")
    email: str = Field(..., max_length=255, description="Email")
    phone: Optional[str] = Field(None, max_length=20, description="電話")
    address: Optional[str] = Field(None, description="地址")
    hire_date: date = Field(..., description="到職日")
    termination_date: Optional[date] = Field(None, description="離職日")
    is_active: bool = Field(True, description="是否啟用")


class EmployeeUpdate(BaseModel):
    """更新員工"""
    employee_type: Optional[str] = Field(None, description="員工類型 (admin/full_time/part_time/intern)")
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    email: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=20)
    address: Optional[str] = None
    hire_date: Optional[date] = None
    termination_date: Optional[date] = None
    is_active: Optional[bool] = None


class EmployeeResponse(BaseModel):
    """員工回應"""
    id: str
    employee_no: str
    employee_type: str
    name: str
    email: str
    phone: Optional[str] = None
    address: Optional[str] = None
    hire_date: Optional[date] = None
    termination_date: Optional[date] = None
    is_active: bool = True
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class EmployeeListResponse(BaseModel):
    """員工列表回應"""
    success: bool = True
    data: list[EmployeeResponse] = []
    total: int = 0
    page: int = 1
    per_page: int = 20
    total_pages: int = 0

File: frontend/src/api/employee.ts
import request from '@/utils/request';
import type { BaseResponse, DataResponse, ListResponse } from './response';

export type EmployeeType = 'admin' | 'full_time' | 'part_time' | 'intern';

export interface EmployeeRole {
  id: string;
  key: string;
  name: string;
}

export interface Employee {
  id: string;
  employee_no: string;
  employee_type: EmployeeType;
  name: string;
  email: string;
  phone?: string | null;
  address?: string | null;
  hire_date?: string | null;
  termination_date?: string | null;
  is_active: boolean;
  has_account: boolean;
  role_id?: string | null;
  role_name?: string | null;
  email_verified_at?: string | null;
  created_at?: string;
  updated_at?: string;
}

export interface EmployeeListParams {
  page?: number;
  per_page?: number;
  search?: string;
  is_active?: boolean;
  employee_type?: EmployeeType;
}

export interface EmployeeCreate {
  employee_no: string;
  employee_type: EmployeeType;
  name: string;
  email: string;
  phone?: string;
  address?: string;
  hire_date: string;
  termination_date?: string;
  is_active: boolean;
}

export interface EmployeeUpdate {
  employee_type?: EmployeeType;
  name?: string;
  email?: string;
  phone?: string;
  address?: string;
  hire_date?: string;
  termination_date?: string;
  role_id?: string;
  is_active?: boolean;
}

export type EmployeeListResponse = ListResponse<Employee>;

export const getEmployeesApi = (params?: EmployeeListParams) => {
  return request.get<any, EmployeeListResponse>('/v1/employees', { params });
};

export const getEmployeeApi = (id: string) => {
  return request.get<any, DataResponse<Employee>>(`/v1/employees/${id}`);
};

export const getEmployeeRolesApi = () => {
  return request.get<any, DataResponse<EmployeeRole[]>>('/v1/employees/roles');
};

export const createEmployeeApi = (data: EmployeeCreate) => {
  return request.post<any, DataResponse<Employee>>('/v1/employees', data);
};

export const updateEmployeeApi = (id: string, data: EmployeeUpdate) => {
  return request.put<any, DataResponse<Employee>>(`/v1/employees/${id}`, data);
};

export const deleteEmployeeApi = (id: string) => {
  return request.delete<any, BaseResponse>(`/v1/employees/${id}`);
};
